### ISSUE TITLE
Melee scaling mode is not properly loaded on startup, causing graphical issues

### ISSUE THREAD

**Issue by Ala-lala** [JHGuitarFreak/UQM-MegaMod-Archived#19]
The melee scaling mode doesn't seem to be properly converted on non-Android platforms when it is initially read from the config file upon startup, making the smooth zooming use nearest-neighbor scaling. This causes very noticeable graphical issues when using the original graphics and smooth zooming.

There are three workarounds that I could find:
* Explicitly specify the zoom mode on the command line with `-b smooth`
* Open and close the setup menu (which properly handles the scaling mode) every time you open UQM
* Delete the `smoothmelee = INT32:1` line in `uqm.cfg` (this must be redone every time you use the setup menu)

**Comment by Serosis** [comment]
Snap. Thanks for pointing this out I'll take a look.

**Comment by Serosis** [comment]
Fixed

**Comment by Ala-lala** [comment]
This doesn't quite work either: `getBoolConfigValueXlat` requires the provided config value to be a boolean, whereas `config.smoothmelee` is an integer (0 or 1 on desktop platforms). This makes it always use the default setting (trilinear smooth scaling) on startup. Using something like:
```
if (res_IsInteger("config.smoothmelee") && !options->meleeScale.set) {
#if defined(ANDROID) || defined(__ANDROID__)
	options->meleeScale.value = res_GetInteger("config.smoothmelee");
#else
	options->meleeScale.value = res_GetInteger("config.smoothmelee") ? TFB_SCALE_TRILINEAR : TFB_SCALE_STEP;
#endif
	options->meleeScale.set = true;
}
```
instead should work.

**Comment by Ala-lala** [comment]
(This is minor, but also, line endings in `uqm.c` are inconsistent now: `#else` and the line below it use CRLF while the rest use LF)

**Comment by Serosis** [comment]
Yeah, that's Visual Studio for you, next time I open it up it will give me a pop-up about it.

**Comment by Serosis** [comment]
Fixed smooth scaling options in commit 6294c27. Overlooked that everything but Android was supposed to be boolean until you said something and that the value being written was always an integer before now.
